app/controllers/api/storage.php

```diff
@@ -245,6 +246,218 @@
+            '$write' => (is_null($write) && !$user->isEmpty()) ? ['user:'.$user->getId()] : $write ?? [], // By default set write permissions for user
+            'dateCreated' => \time(),
+            'bucketId' => $bucket->getId(),
+            'name' => $file['name'],
+            'path' => $path,
+            'signature' => $device->getFileHash($path),
+            'mimeType' => $mimeType,
+            'sizeOriginal' => $size,
+            'sizeActual' => $sizeActual,
+            'algorithm' => empty($compressor) ? '' : $compressor->getName(),
+            'comment' => '',
+        ];
+
+        if($bucket->getAttribute('encryption', true)) {
+            $data['openSSLVersion'] = '1';
+            $data['openSSLCipher'] = OpenSSL::CIPHER_AES_128_GCM;
+            $data['openSSLTag'] = \bin2hex($tag);
+            $data['openSSLIV'] = \bin2hex($iv);
+        }
+
+        $file = $dbForInternal->createDocument('files', new Document($data));
+
+        $audits
+            ->setParam('event', 'storage.files.create')
+            ->setParam('resource', 'storage/files/'.$file->getId())
+        ;
+
+        $usage
+            ->setParam('storage', $sizeActual)
+        ;
+
+        $response->setStatusCode(Response::STATUS_CODE_CREATED);
+        $response->dynamic2($file, Response::MODEL_FILE);
+        ;
+    });
+
+App::get('/v1/storage/buckets/:bucketId/files')
+    ->desc('List Files')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'files.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'listFiles')
+    ->label('sdk.description', '/docs/references/storage/list-files.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_FILE_LIST)
+    ->param('bucketId', null, new UID(), 'Storage bucket unique ID. You can create a new storage bucket using the Storage service [server integration](/docs/server/storage#createBucket).')
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->param('limit', 25, new Range(0, 100), 'Results limit value. By default will return maximum 25 results. Maximum of 100 results allowed per request.', true)
+    ->param('offset', 0, new Range(0, 2000), 'Results offset. The default value is 0. Use this param to manage pagination.', true)
+    ->param('orderType', 'ASC', new WhiteList(['ASC', 'DESC'], true), 'Order result by ASC or DESC order.', true)
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->action(function ($bucketId, $search, $limit, $offset, $orderType, $response, $dbForInternal) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
```

Comment:
We should add 404 check for the bucket before we read the files, same as we do for collections. This is true for all /files endpoints